experiment/contractNipopow.sol

```diff
@@ -78,12 +79,27 @@ contract Crosschain {
             uint256 pow = (i / 8) * 8 + 8 - (i % 8) - 1;
             uint256 mask = 2**pow;
             if ((hash & mask) != 0) {
-                return uint8(i);
+                return uint8(i) - 1;
             }
         }
         return 0;
     }
 
+    function bestArgAtLevel(bytes32[] memory proof, uint256 level)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 bestLevelCounter = 0;
```

Comment:
should this be called bestLevelCounter? there's no *best* here, there's a specific level you're counting blocks in

Reply:
Done